articles/modules/ROOT/pages/how-to-force-direction-of-expansion.adoc

```diff
@@ -0,0 +1,128 @@
+
+In the case of more complicated queries, there may be quite a few different nodes that could potentially be used as anchor nodes, with many possibilities on which one or which combination to anchor on, and how to expand to fulfill the desired patterns.
+
+In any case, it is possible for the planner to make a bad choice, either because the approach isn't universally efficient across all data in your graph (some nodes may be supernodes, and cause the query over them to choke) or the metadata available to the planner is insufficient to warn it away from these more expensive expansions.
+
+In these cases, what we as humans know about the general shape of the graph may be greater than what can be inferred via metadata. Remember that walking the actual graph data is not possible here, since we're talking about query planning, which precedes execution.
+
+== Ways to force the direction of an expansion
+
+Unfortunately we do not yet have planner hints that directly require or forbid expansion in a certain direction.
+Instead, we must influence this indirectly, through other hints, or through Cypher tricks which leave the planner no other choice.
+
+=== Using hash joins to force expansion to a supernode
+
+Remember that supernodes are really only problematic when expanding through, or expanding away from, but depending on your graph data it may be just fine if you are only expanding to them.
+
+
+In the case where you have multiple efficient anchor nodes, and a known or potential supernode in the middle of the pattern, and you know the expansion TO the supernode from both sides is cheap, you can use a join hint to force expanding to the super node.
+
+This is discussed in more detail on an alternate article here:
+
+https://neo4j.com/developer/kb/how-to-avoid-costly-traversals-with-join-hints/[How to avoid costly traversals with join hints]
+
+=== Label filtering without anchoring
+
+The presence of a label on a node in the Match pattern can significantly change whether or not the planner will try to use it as an anchor node.
+
+That is, the planner prefers index lookups to find anchor nodes, though it may still use label scans to find anchor nodes (especially when there are no opportunities for index lookups in the query).
+
+Neither index lookups or label scans are possible when the label isn't present in the pattern (or otherwise described in a Where clause, like `WHERE c:Country`), and these two means of lookup are the two most common for finding anchor nodes in the graph.
+Of course, you may need to have that label in the query for the sake of correctness, otherwise the wrong nodes and patterns might be matched.
+
+There is a way to perform the label filtering in a special manner which also prevents the planner from using it for index lookup or label scanning.
+Here is an example:
+
+[source,cypher]
+----
+MATCH (user:User {id:12345})-[:LIVES_IN]->(country {name:'United States'})
```

Comment:
This is really ouch, not sure you want to show this.
in this case not checking the label should be fine as the rel-type is specific enough.